enterprise/views.py

```diff
@@ -771,7 +771,16 @@ def post(self, request, enterprise_uuid, course_id):
                 ),
                 query_string=urlencode({'course_mode': selected_course_mode_name})
             )
-            failure_url = reverse('enterprise_course_enrollment_page', args=[enterprise_customer.uuid, course_id])
+            if request.META['QUERY_STRING']:
```

Comment:
Will this work? Seems like you would need to include the query string with the POST request. It doesn't look like we have made that change here.

Reply:
@douglashall This fix is working you can verify this on business sandbox. The catalog uuid is provided as querystring in the URL and this URL in turn will be used to construct `failure_url` on which a GET request is sent. Since this is POST method so I used `request.META` instead of `request.GET` to get the querystring. This way all the querystring parameters will automatically preserved instead of updating the course enrollment page to send only `catalog` field and then extract and use it here.
Here is the complete URL when a learner clicks on the `Continue` button and DSC is required.

http://testserver/enterprise/grant_data_sharing_permissions?course_id=course-v1:edX+DemoX+Demo_Course&failure_url=/enterprise/489308cc-e4ed-6469-9a91-eaa2f57c4cb4/course/course-v1:edX+DemoX+Demo_Course/enroll/catalog=f2ed8299-b0f0-4d55-ab2d-5a2919b34dc3&enterprise_customer_uuid=489308cc-e4ed-6469-9a91-eaa2f57c4cb4&next=/enterprise/handle_consent_enrollment/489308cc-e4ed-6469-9a91-eaa2f57c4cb4/course/course-v1:edX+DemoX+Demo_Course/?course_mode=audit

Reply:
Ah, I see now. Thanks for the clarification.